### PR TITLE
SIDM-6598 - Added editing user details functionality

### PR DESCRIPTION
**Pull request by Thomas-Geraghty**
### JIRA link ###

https://tools.hmcts.net/jira/browse/SIDM-6598

### Change description ###

- Added IdamApi methods to:
  - get all roles 
  - update a users details by ID
  - update a users roles by ID
- Added UserActionsController to handle editing users, and in future, deleting, suspending users.
- Added edit-user views and custom styling

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/app/idam-api/IdamAPI.ts [hunk 1]
```diff
@@ -20,4 +20,27 @@ export class IdamAPI {
         return [];
       });
   }
+
+  public getUserById(id: string): Promise<User> {
```
Comment (KianKwa) [inline src/main/app/idam-api/IdamAPI.ts:24]: Do we need this method? We have the getUserDetails() method which calls out to the IDAM API to either retrieve email, ID or SSO ID.

File: src/main/controllers/UserActionsController.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    return changedFields;
+  }
+
+  private validateFields(fields: Partial<User>): PageError {
+    const { forename, surname, email } = fields;
+    const errors: any = {};
+
+    if(hasProperty(fields, 'forename') && isEmpty(forename)) errors.forename = { message: USER_EMPTY_FORENAME_ERROR };
+    if(hasProperty(fields, 'surname') &&  isEmpty(surname)) errors.surname = { message: USER_EMPTY_SURNAME_ERROR };
+    if(hasProperty(fields, 'email') && !isValidEmailFormat(email)) errors.email = {message: INVALID_EMAIL_FORMAT_ERROR };
```
Comment (KianKwa) [inline src/main/controllers/UserActionsController.ts:91]: The search for user page currently has 2 email errors (one if the email is blank and another one for invalid email format). It is not a big deal but up to you whether you want to check for 2 separate errors or encompassing both in the same message.

**Review by KianKwa — APPROVED** [review]

File: src/main/app/idam-api/IdamAPI.ts (hunk 1 above)
Comment (Thomas-Geraghty) [reply src/main/app/idam-api/IdamAPI.ts:24]: As discussed, will address this in a future PR :)

File: src/main/controllers/UserActionsController.ts (hunk 2 above)
Comment (Thomas-Geraghty) [reply src/main/controllers/UserActionsController.ts:93]: Resolved in https://github.com/hmcts/idam-user-dashboard/pull/209/commits/d391c7d8df36352242935943048aa6e85c9f49da

**Comment by Thomas-Geraghty** [comment]
Reviewed @ashwini-mv functional testing, looks all good. :)

**Pull request merged**